controllers/vspherecluster_reconciler.go

```diff
@@ -151,37 +147,8 @@ func (r *clusterReconciler) reconcileDelete(ctx context.Context, clusterCtx *cap
 			"unable to list VSphereMachines part of VSphereCluster %s/%s", clusterCtx.VSphereCluster.Namespace, clusterCtx.VSphereCluster.Name)
 	}
 
-	machineDeletionCount := 0
-	var deletionErrors []error
-	for _, vsphereMachine := range vsphereMachines {
-		// Note: We have to use := here to not overwrite log & ctx outside the for loop.
-		log := log.WithValues("VSphereMachine", klog.KObj(vsphereMachine))
-		ctx := ctrl.LoggerInto(ctx, log)
-
-		// If the VSphereMachine is not owned by the CAPI Machine object because the machine object was deleted
-		// before setting the owner references, then proceed with the deletion of the VSphereMachine object.
-		// This is required until CAPI has a solution for https://github.com/kubernetes-sigs/cluster-api/issues/5483
-		if !clusterutilv1.IsOwnedByObject(vsphereMachine, clusterCtx.VSphereCluster) || len(vsphereMachine.OwnerReferences) != 1 {
-			continue
-		}
-		machineDeletionCount++
-		// Remove the finalizer since VM creation wouldn't proceed
-		log.Info("Removing finalizer from VSphereMachine")
-		ctrlutil.RemoveFinalizer(vsphereMachine, infrav1.MachineFinalizer)
```

Comment:
Don't we still need something which removes this finalizer? Or should this already have happened a longer time ago?

Reply:
You're right - let me put this on [WIP] - I'm still trying to test it properly.

Reply:
This finalizer is going to be removed in the vspheremachine_controller at https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/959691ca530bc1972092d1a9233c4029b831b289/controllers/vspheremachine_controller.go#L285.

Previously the VSphereCluster was able to do a full cleanup in case VSphereMachines were orphaned. They shouldn't be able to become orphaned now as CAPI ensure InfrastructureMachines should always have an ownerReference.